app/src/js/components/Conversations/conversation.js

```diff
@@ -20,66 +20,221 @@ import { AddAttachmentButton, DisplayAttachmentButton } from './attachment';
+const Conversation = ({ dispatch, conversation, privileges, match, user }) => {
   const current_user_id = JSON.parse(window.localStorage.getItem('auth-user')).id;
   const { conversationId } = match.params;
   const [showSearch, setShowSearch] = useState(false);
+  const [level, setLevel] = useState(false);
   const [uploadedFiles, setUploadedFiles] = useState([]);
-  useEffect(() => {
-    dispatch(getConversation(conversationId));
-  }, []);
+  // useEffect(() => {
+  //   dispatch(getConversation(conversationId));
+  // }, []);
+  const getConversations = async(dispatch, conversationId, lvl) => {
+    if (lvl) {
+      setLevel(lvl);
+      document.getElementById('all_button').classList.add('active');
+      document.getElementById('users_only_button').classList.remove('active');
+    } else {
+      setLevel(false);
+      document.getElementById('all_button').classList.remove('active');
+      document.getElementById('users_only_button').classList.add('active');
+    }
+
+    const newData = await dispatch(getConversation(conversationId, lvl));
+    if(newData?.data?.notes) setDisplayNotes(newData?.data?.notes);
+  };
   const { data, inflight, meta } = conversation;
-  const { subject, notes, participants } = data;
+  const { subject, notes = [], participants = [] } = data;
   const { queriedAt } = meta;
   const { canReply, canAddUser } = notePrivileges(privileges);
   const visibilityRef = useRef();
   const uploadedFilesRef = useRef();
 
+  const [tempNotes, setTempNotes] = useState([]);
+  const [displayNotes, setDisplayNotes] = useState(data.notes);
+  const [shouldStopRetries, setShouldStopRetries] = useState(false);
+
   const handleVisibilityReset = () => {
     visibilityRef?.current?.resetIdMap();
   };
 
+  const MAX_RETRIES = 7;
+  // Retry at 3s, 10s, 20s, 35s, 50s, 65s, 80s
+  const RETRY_INTERVALS = [3000, 5000, 10000, 15000, 15000, 15000, 15000];
+  let currentTimeout = null; // Store timeout ID
+
+  useEffect(() => {
+    const fetchNotes = async () => {
+        const finalNotes = await dispatch(getConversation(conversationId, level));
+        const updatedNotes = finalNotes?.data?.notes.map(note => ({
+            ...note,
+            isPendingAttachmentMatch: note.isPendingAttachmentMatch ?? false
+        }));
+
+        setDisplayNotes(updatedNotes);
+    };
+
+    if (shouldStopRetries) return;
+
+    const firstNewNote = notes[0];
+    const firstTempNote = tempNotes.length > 0 ? tempNotes[0] : null;
+
+    if (!firstTempNote) {
+        setDisplayNotes(notes);
+        return;
+    }
+
+    const isTextMatch = firstTempNote.text.trim() === firstNewNote?.text?.trim();
+    const areAttachmentsMatch =
+        new Set(firstTempNote.attachments).size === new Set(firstNewNote?.attachments || []).size &&
+        firstTempNote.attachments.every(att =>
+            (firstNewNote?.attachments || []).some(newAtt => newAtt.trim() === att.trim())
+        );
+
+    if (isTextMatch && areAttachmentsMatch) {
```

Comment:
One thing I noticed is there is an edge case here where the note has not been created yet on the backend before the first page refresh, so `isTextMatch` is false here and doesn't actually attempt any other refreshes. This may just have to be an accepted edge case, though, because the only solution I can think of would be sending the note id from the dashboard and comparing that instead of the text; however, that extends beyond the scope of this task. Maybe something to remember for a backlog task.